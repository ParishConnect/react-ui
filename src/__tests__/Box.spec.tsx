import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import "jest";
import Box from "../Box/Box";
import { PropDefs } from "../Box/PropDefs";

test("Box Renders Correctly", () => {
	const tree: any = renderer.create(<Box />).toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree).toHaveStyleRule("width", "10px");
	expect(tree).toHaveStyleRule("height", "10px");
	expect(tree).toHaveStyleRule("display", "block");
});

test("PropDefs is Correct format", () => {
	expect(PropDefs).toBeInstanceOf(Array);
	PropDefs.map((def: { prop: string; type: string }) => {
		expect(def).toHaveProperty("prop");
	});
});
