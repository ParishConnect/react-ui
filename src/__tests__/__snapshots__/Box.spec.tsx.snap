// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Renders Correctly 1`] = `
.c0 {
  height: 10px;
  width: 10px;
  display: block;
}

<div
  className="c0"
  display="block"
  height={10}
  width={10}
/>
`;
